packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -2789,7 +2803,33 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
       onLiveTextInput: liveTextInputEnabled
           ? () => _startLiveTextInput(SelectionChangedCause.toolbar)
           : null,
-    );
+    )..addAll(_textProcessingActionButtonItems);
+  }
+
+  List<ContextMenuButtonItem> get _textProcessingActionButtonItems {
+    final List<ContextMenuButtonItem> buttonItems = <ContextMenuButtonItem>[];
+
+    if (widget.obscureText || textEditingValue.selection.isCollapsed) {
+      return buttonItems;
+    }
+
+    for (final ProcessTextAction action in _processTextActions) {
+      buttonItems.add(ContextMenuButtonItem(
+        label: action.label,
+        onPressed: () async {
+          final String selectedText = textEditingValue.selection.textInside(textEditingValue.text);
+          if (selectedText.isNotEmpty) {
+            final String? processedText = await _processTextService.processTextAction(action.id, selectedText, widget.readOnly);
```

Comment:
For my understanding what does an app developer expect when the system wide button is used  on static text?

Reply:
If the buttons are expected to do nothing on static text, do they still show up when used on static text despite that?

Reply:
They are two types of activities registered to Intent.PROCESS_TEXT:
- some will use the text (`selectedText` variable in this code) to open their UI and will display a result (for instance, on my phone I have one activity that will read the text loudly, I also have Google Translate and Deepl that will show the translation using their own panel). Such activities does not return a result (`processedText` will be null in this case).
- some will use the selected text to return a modified version (on my phone I have 'Encrypt Text' that does that, it opens an UI where I can ask to see the encrypted text and a button to replace the current selection with this encrypted text). Such activities returns a result (`processedText` won't be null in this case and we use it to replace the current selection).